infrastructure/templates/public-api/main.bicep

```diff
@@ -185,32 +187,19 @@ module postgreSqlServerModule 'components/postgresqlDatabase.bicep' = {
     dbAutoGrowStatus: postgreSqlAutoGrowStatus
     postgreSqlVersion: '16'
     tagValues: tagValues
-    privateDnsZoneId: postgreSqlPrivateDnsZone.id
-    firewallRules: postgreSqlFirewallRules
-    subnetId: vNetModule.outputs.postgreSqlSubnetRef
+    firewallRules: formattedPostgreSqlFirewallRules
     databaseNames: ['public_data']
   }
 }
 
-resource apiContainerAppManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
-  name: apiContainerAppManagedIdentityName
-  location: location
-}
-
-var acrPullRole = resourceId('Microsoft.Authorization/roleDefinitions', '7f951dda-4ed3-4680-a7ca-43fe172d538d')
+var psqlManagedIdentityConnectionStringTemplate = 'Server=${psqlServerFullName}.postgres.database.azure.com;Database=[database_name];Port=5432;User Id=[managed_identity_name];Password=[access_token]'
 
-@description('This allows the managed identity of the container app to access the registry, note scope is applied to the wider ResourceGroup not the ACR')
-resource apiContainerAppManagedIdentityRBAC 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
-  name: guid(resourceGroup().id, apiContainerAppManagedIdentity.id, acrPullRole)
-  properties: {
-    roleDefinitionId: acrPullRole
-    principalId: apiContainerAppManagedIdentity.properties.principalId
-    principalType: 'ServicePrincipal'
-  }
+resource apiContainerAppManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = if (deployContainerApp) {
+  name: apiContainerAppManagedIdentityName
 }
```

Comment:
Could this be inside the containerApp module rather than in main?

Reply:
I've created https://dfedigital.atlassian.net/browse/EES-5128 to cover this, and added a TODO to the codebase to reference it.